### PR TITLE
[-] BO : introduced a reusable way to display a right sidebar

### PR DESCRIPTION
**Pull request by mickaelandrieu**
## Description

We want to be able to act on a right sidebar to display related informations.
For instance, on product catalog we can quickly move to another product throught the product list sidebar.

quick_nav

We want to use this system to make the docs always available anywhere, and related to the page the e-merchandizer actualy visits. (like in 1.6)

The new migrated pages have an inconsistent behavior on this point: this contribute is about put the docs the right way, using a reusable sidebar system.
## Steps to Test this Fix
1. Step 1.
   Visit all Product/Module pages
2. [x] _Product_ catalog
3. [x] _Product_ form
4. [x] _Modules_ catalog
5. [x] _Modules_ installed modules
6. [x] _Modules_ notifications
7. Step 2.
   Click on the help button in the top bar
8. Step 3.
   You should see a sidebar displayed

products_help
## Forge Ticket (optional)

This improvement will fix http://forge.prestashop.com/browse/BOOM-238 and make the Back Office easier to maintain.

### CODE REVIEW

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 1]
```diff
@@ -1945,26 +1945,28 @@ var rightSidebar = (function() {
 	return {
 		'init': function() {
 			/** Right sidebar should shrink only step1 tab content */
```
Comment (mickaelandrieu) [inline admin-dev/themes/default/js/bundle/product/form.js]: note to myself: remove this comment.

File: admin-dev/themes/new-theme/scss/theme.scss [hunk 2]
```diff
@@ -73,3 +73,8 @@ $color-icons: #6c868e;
   float:none;
   margin:0 auto;
 }
+
+.fullspace {
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/scss/theme.scss]: This is an helper to force the docs page loaded in an HTML5 <object> to take all the available space.

File: src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php [hunk 3]
```diff
@@ -28,6 +28,7 @@
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\Form\Form;
 use PrestaShop\PrestaShop\Adapter\Configuration;
+use Symfony\Component\BrowserKit\Client;
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php]: unused use statment.

File: src/PrestaShopBundle/Resources/views/Admin/layout.html.twig [hunk 4]
```diff
@@ -12,6 +12,8 @@
 {% import 'PrestaShopBundle:Admin:macros.html.twig' as ps %}
 
 {% block stylesheets %}
+    {{ parent() }}
+    <link rel="stylesheet" href="{{ asset('themes/default/css/bundle/product/right-sidebar.css') }}" />
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/layout.html.twig]: note to myself: this css shoud be moved to `/css/bundle/` instead

File: src/PrestaShopBundle/Twig/LayoutExtension.php [hunk 5]
```diff
@@ -179,7 +187,7 @@ public function getLegacyLayout($controllerName = "", $title = "", $headerToolba
      */
     public function getAdminLink($controllerName, $withToken = true, $extraParams = [])
     {
-        return $this->context->getAdminLink($controllerName, $withToken = true, $extraParams = []);
+        return $this->context->getAdminLink($controllerName, $withToken, $extraParams);
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/LayoutExtension.php]: should be done in another pull request.

File: src/PrestaShopBundle/Controller/Admin/CommonController.php [hunk 6]
```diff
@@ -147,7 +147,7 @@ public function paginationAction(Request $request, $limit = 10, $offset = 0, $to
             'limit_choices' => $limitChoices,
         );
         if ($view != 'full') {
-            return $this->render('PrestaShopBundle:Admin:Common/pagination_'.$view.'.html.twig', $vars);
+            return $this->render('PrestaShopBundle:Admin/Common/pagination_'.$view.'.html.twig', $vars);
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/CommonController.php]: should be done in another pull request (pagination in 'grid' view don't works as expected).

File: admin-dev/themes/new-theme/scss/theme.scss (hunk 2 above)
Comment (Shudrum) [inline admin-dev/themes/new-theme/scss/theme.scss]: Hey! Thank you!

As we try to follow the RSCSS norm, please add, for the helpers, an underscore : `_fullspace`.

http://rscss.io/

:+1:

**Comment by kpodemski** [comment]
I like this idea but it will be nice to have both: help sidebar and "some sidebar", and if we - as module developers - will have possibility to use it in our Controllers it will be great, but I think it should have dedicated icon

**Comment by mickaelandrieu** [comment]
@kpodemdki this right bar is totally reusable, the help button will only set the header title to "Documentation". 

If some modules enjoy this feature for new modules in 1.7,  it's fine to me ;)

File: admin-dev/themes/new-theme/scss/theme.scss (hunk 2 above)
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/scss/theme.scss]: updated, thank you :)

**Comment by Shudrum** [comment]
:+1:

Purrfect!  🐱

**Comment by mickaelandrieu** [comment]
^^ lets wait for @vincentbz functional validation  ;)
